File: Python/SameTree.py
# -*- coding: UTF-8 -*-
# Given two binary trees, write a function to check if they are equal or not.
#
# Two binary trees are considered equal if they are structurally identical and the nodes have the same value.
#
# Python, Python 3 all accepted.


class SameTree(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        if p is None and q is None:
            return True
        if p is None or q is None:
            return False

        while True:
            return p.val == q.val and self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)


# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

File: JavaScript/src/SameTree.js
/**
 * Given two binary trees, write a function to check if they are equal or not.
 * Two binary trees are considered equal if they are structurally identical and the nodes have the same value.
 *
 * Accepted.
 */

function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function (p, q) {
    return p === null && q === null || p !== null && q !== null && p.val === q.val && isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
};

if (isSameTree(null, null)) {
    console.log("pass")
} else {
    console.error("failed")
}

var tn1 = new TreeNode(1);
if (isSameTree(null, tn1) === false) {
    console.log("pass")
} else {
    console.error("failed")
}
if (isSameTree(tn1, null) === false) {
    console.log("pass")
} else {
    console.error("failed")
}

var tn2 = new TreeNode(2);
if (isSameTree(tn1, tn2) === false) {
    console.log("pass")
} else {
    console.error("failed")
}

var tn1111 = new TreeNode(1);
tn1111.left = new TreeNode(1);
tn1111.right = new TreeNode(1);
var tn2111 = new TreeNode(1);
tn2111.left = new TreeNode(1);
tn2111.right = new TreeNode(1);
if (isSameTree(tn1111, tn2111)) {
    console.log("pass")
} else {
    console.error("failed")
}